### PR TITLE
Required fields, and form separation

### LINKED ISSUE #43: Warn about incomplete report fields while allowing to be submitted
Location, town, and occurred_at (which should be moved back to IncidentReport) should be required.

### LINKED ISSUE #57: reorganize forms
Different pages or tabs to avoid confusion with saving

### PR DESCRIPTION
**Pull request by dfaulken**
Closes #43. Closes #57.

Required fields:
screen shot 2017-12-14 at 11 24 57 am

Staff view of editing an incident:
screen shot 2017-12-14 at 11 32 52 am

Supervisor view of editing an incident:
screen shot 2017-12-14 at 11 34 49 am

### CODE REVIEW

File: app/controllers/incident_reports_controller.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def update
       else redirect_to @incident,
                   notice: 'Incident report was successfully saved.'
       end
-    else render 'incidents/edit'
+    else render 'edit'
```
Comment (mboneil10) [inline app/controllers/incident_reports_controller.rb]: Why does `incidents` not matter anymore? Rails stuff?

File: app/models/incident_report.rb [hunk 2]
```diff
@@ -110,6 +112,13 @@ def other?
     !(motor_vehicle_collision? || passenger_incident?)
   end
 
+  def report_type
```
Comment (mboneil10) [inline app/models/incident_report.rb]: Nice

File: app/controllers/incident_reports_controller.rb (hunk 1 above)
Comment (dfaulken) [reply app/controllers/incident_reports_controller.rb]: Previously `incidents/edit` contained the edit forms for the incident, the driver account of incident, the supervisor account of incident, and the supervisor report. This PR breaks each of them off into their own edit page.

For this reason, we don't want to always send people back to the top-level page anymore.

File: app/views/incidents/edit.html.haml [hunk 3]
```diff
@@ -9,24 +9,20 @@
...
+  %hr
   %h2 Incident Details
   = render 'incidents/form', incident: @incident
-- if @current_user == @incident.driver || @current_user.staff?
-  %hr
-  %h2 Driver Report
-  = render 'incident_reports/form',
-    report: @incident.driver_incident_report
 %hr
+- if @current_user == @incident.driver || @current_user.staff?
```
Comment (mboneil10) [inline app/views/incidents/edit.html.haml]: Just a question about incidents: can anyone who is a staff edit the driver account of an incident?

File: app/controllers/incident_reports_controller.rb (hunk 1 above)
Comment (mboneil10) [reply app/controllers/incident_reports_controller.rb]: OHHH

Comment (mboneil10) [reply app/controllers/incident_reports_controller.rb]: Yo sé.

File: app/views/incidents/edit.html.haml (hunk 3 above)
Comment (mboneil10) [reply app/views/incidents/edit.html.haml]: (If I understand this properly)

Comment (dfaulken) [reply app/views/incidents/edit.html.haml]: Correct.

**Comment by mboneil10** [comment]
I'm also guessing that in the first screenshot the town and location fields will also be surrounded in red?

**Comment by dfaulken** [comment]
> I'm also guessing that in the first screenshot the town and location fields will also be surrounded in red?

Correct.

**Comment by mboneil10** [comment]
Sorry. I can't English @dfaulken. LGTM!

File: app/controllers/incident_reports_controller.rb (hunk 1 above)
Comment (dfaulken) [reply app/controllers/incident_reports_controller.rb]: Usted sabe? O usted ve? 😛

**Pull request merged**